### PR TITLE
VR-131: profile icons.

### PR DESCRIPTION
**Pull request by n3op2**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [x] Feature

## Linked tickets

[
](https://digicatapult.atlassian.net/jira/software/c/projects/ENG/boards/206?selectedIssue=VR-131)

## High level description

Adding a profile icon to the side bar.

## Detailed description

Please refer to the screenshots

Screenshot 2024-09-06 at 08 34 08
Screenshot 2024-09-06 at 08 33 44

## Operational impact

none, a new item

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -81,6 +81,8 @@ services:
       - LOG_LEVEL=trace
       - DB_HOST=postgres-veritable-ui-bob
       - DB_NAME=veritable-ui
+      - API_SWAGGER_TITLE=Bob
+      - API_SWAGGER_BG_COLOR=#ff3131
```
Comment (n3op2) [inline docker-compose.yml:85]: As per story I was no able to find `LABEL`, only in cloudagent. Used API_TITLE and API_BG_COLOR instead

File: src/env.ts [hunk 2]
```diff
@@ -57,8 +64,8 @@ export const envConfig = {
   DB_PORT: envalid.port({ default: 5432 }),
   COOKIE_SESSION_KEYS: strArrayValidator({ devDefault: ['secret'] }),
   PUBLIC_URL: envalid.url({ devDefault: 'http://localhost:3000' }),
-  API_SWAGGER_BG_COLOR: envalid.str({ default: '#fafafa' }),
-  API_SWAGGER_TITLE: envalid.str({ default: 'Veritable' }),
+  API_SWAGGER_BG_COLOR: envalid.str({ devDefault: '#ff3131', default: '#ffffff' }),
+  API_SWAGGER_TITLE: envalid.str({ devDefault: 'Bob', default: 'veritable' }),
```
Comment (dblane-digicatapult) [inline src/env.ts:68]: Why add these devDefaults?

Comment (n3op2) [reply src/env.ts:68]: for demo:) since demo mode default is true, and we setup as alice initially anyway

Comment (n3op2) [reply src/env.ts]: updated, thanks for a flag up.

Comment (Ellenn-A) [reply src/env.ts]: so this is the one that's meant to be 'Alice'?

File: src/views/common.tsx [hunk 3]
```diff
@@ -55,9 +57,15 @@ export const FormButton = (props: FormButtonProps): JSX.Element => (
  * @returns JSX - Sidarbar
  */
 const SideBar = ({ activePage }: { activePage: PageProps['activePage'] }): JSX.Element => {
+  const title = new Env().get('API_SWAGGER_TITLE').slice(0, 1)
```
Comment (mattdean-digicatapult) [inline src/views/common.tsx:60]: don't pull in `env` like this. Use `container.resolve`

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

File: src/views/common.tsx (hunk 3 above)
Comment (n3op2) [reply src/views/common.tsx]: thanks for a flag up, updated to use `const env = container.resolve(Env)`

File: src/env.ts (hunk 2 above)
Comment (n3op2) [reply src/env.ts]: Indeed, has been updated

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**